trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/credentials/StandardIdentity.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.spi.credentials;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public record StandardIdentity(String user, Optional<List<String>> groups)
```

Comment:
Could we add the groups to the standard `Identity` interface? It feels like it is a generic enough piece of information that making it implementation dependent could be painful

Reply:
What do you think about making `groups` be `List<String>`? We'd need to configure jackson to default the field to an empty list

That may be a bit annoying as IIRC that's done with an annotation (which would make Jackson be a part of the SPI) or we'd need to modify the object mapper configs

Reply:
It's ugly either way. Let me think about it.

Reply:
We can bind a custom Jackson Module in the main code base which would keep Jackson out of the SPI. But, then, we're relying on this custom module. Maybe a reasonable compromise. wdyt?

Reply:
done - PTAL